doc: contributing: Make instructions cross-platform and document CI

carlescufi opened the pull request:
The contributing instructions were currently centered exclusively around
Linux. Make them generic and cross-plaform while highlighting the tools
that do not currently work on Windows, namely:

 * sanitycheck
 * checkpatch

Additional document CI for contributors.

Fixes #5717
Fixes #4772

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
doc: contributing: Make instructions cross-platform

The contributing instructions were currently centered exclusively around
Linux. Make them generic and cross-plaform while highlighting the tools
that do not currently work on Windows, namely:

 * sanitycheck
 * checkpatch

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

dbkinder commented on CONTRIBUTING.rst:
```diff
@@ -143,19 +140,25 @@ and how to set up your development environment as introduced in the Zephyr
+.. note::
+   The Zephyr development workflow supports all 3 major operating systems
+   (Linux, macOS and Windows) but some of the tools used in the sections below
+   are only available on Linux and macOS. On Windows you will need to rely on
```
How about:

    On Windows, instead of running these tools yourself, you will need to rely on
    the Continuous Integration (CI) service ``shippable``, which runs automatically
    on GitHub when you submit your Pull Request (PR).  You can see any failure
    results in the Shippable details link near the end of the PR conversation list.

(As an aside, issue #4772 mentions we don't document how to find and interpret these shippable reports.)

dbkinder commented on CONTRIBUTING.rst:
```diff
@@ -259,8 +270,11 @@ standards <Coding Style>`_ together with a configuration file we've provided:
+   # On Linux/macOS
    uncrustify --replace --no-backup -l C -c
-     $ZEPHYR_BASE/scripts/uncrustify.cfg   my_source_file.c
+     $ZEPHYR_BASE/scripts/uncrustify.cfg my_source_file.c
```
need a backslash on the previous line to indicate a command continuation (or join the lines as you did for the Windows example).

carlescufi committed:
doc: contributing: Document CI

Add basic CI documentation to help users navigate through the CI system
and deal with failures.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
@dbkinder I addressed your comments and also added a CI section in order to close #4772

carlescufi commented:
@dbkinder build fixed, comments addressed

dbkinder commented on CONTRIBUTING.rst:
```diff
@@ -201,32 +207,68 @@ business days.
+
+* Git commit formatting
+* Coding Style
+* Sanity Check builds for multiple architectures and boards
```
Let's add one more:

    * Documentation build to verify any doc changes

carlescufi replied:
Fixed

dbkinder commented on CONTRIBUTING.rst:
```diff
@@ -201,32 +207,68 @@ business days.
+* Sanity Check builds for multiple architectures and boards
+
+CI is run on the ``shippable`` cloud service and it uses the same tools
+described in the `Contribution Tools`_ section. The CI result must be green in
```
How about:

    The CI results must be green indicating "All checks have passed" before
    the Pull Request can be merged.  CI is run when the PR is created, and 
    again every time the PR is modified with a commit.  You can also force
    the CI system to recheck a PR by adding a comment to the PR saying
    simply ``retest`` in the message (helpful if the CI system fails unexpectedly).

carlescufi replied:
Fixed

dbkinder commented on CONTRIBUTING.rst:
```diff
@@ -201,32 +207,68 @@ business days.
+Once you click on the link you will be taken to the ``shippable`` summary
+results page where a table with all the different builds will be shown. To see
+what build or test failed click on the row that contains the failed (i.e.
+non-green) build and then click on the "Tests" tab to see the failures.
```
to see the console output messages indicating the failure.

carlescufi replied:
Fixed

dbkinder reviewed (approved):
thanks
